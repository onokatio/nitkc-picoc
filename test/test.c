int main()
{
	int	i;

	i = - - 1;
	write(i);
	writeln();
}
