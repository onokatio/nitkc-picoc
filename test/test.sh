ls *.c | while read line;do ../test_parser < $line > out/$(echo $line | sed -e 's/\.c//').out ;done
ls *.c | while read line;do ../test_parser_answer < $line > sample_out/$(echo $line | sed -e 's/\.c//').out ;done
ls out | while read line;do echo $line;diff out/$line sample_out/$line;done
