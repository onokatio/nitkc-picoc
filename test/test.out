#Picoc09 2011-12-13
__start__:
	call	main
	halt
read:
	enter
	rd
	leave
	ret
write:
	enter
	pushl 2
	wr
	leave
	ret
#組込み関数 writeln()
writeln:
	enter
	wrln
	leave
	ret
main:
	enter
	mvsp	-1
	pushi	1
	pushi	-1
	mul
	pushi	-1
	mul
	storel	-1
	mvsp	1
	mvsp	-1
	call	write
	storet	1
	mvsp	1
	mvsp	1
	mvsp	-1
	call	writeln
	mvsp	1
	mvsp	1
	leave
	ret
